tests/integration/ha_tests/helpers.py

```diff
@@ -488,3 +519,116 @@ async def verify_writes(ops_test: OpsTest) -> int:
             total_expected_writes == actual_writes
         ), f"{role} {unit.name} missed writes to the db."
     return total_expected_writes
+
+
+async def get_other_mongodb_direct_client(ops_test: OpsTest, app_name: str) -> MongoClient:
+    """Returns a direct mongodb client to the second mongodb cluster."""
+    unit = ops_test.model.applications[app_name].units[0]
+    action = await unit.run_action("get-password")
+    action = await action.wait()
+    password = action.results["operator-password"]
+    status = await ops_test.model.get_status()
+    address = status["applications"][app_name]["units"][unit.name]["address"]
+
+    return MongoClient(f"mongodb://operator:{password}@{address}/admin", directConnection=True)
+
+
+def retrieve_entries(client, db_name, collection_name, query_field):
+    """Retries entries from a specified collection from a provided client."""
+    db = client[db_name]
+    test_collection = db[collection_name]
+
+    # read all entries from original cluster
+    cursor = test_collection.find({})
+    cluster_entries = set()
+    for document in cursor:
+        cluster_entries.add(document[query_field])
+
+    return cluster_entries
+
+
+def deploy_chaos_mesh(namespace: str) -> None:
+    """Deploy chaos mesh to the provided namespace.
+
+    Args:
+        namespace: The namespace to deploy chaos mesh to
+    """
+    env = os.environ
+    env["KUBECONFIG"] = os.path.expanduser("~/.kube/config")
+
+    subprocess.check_output(
+        " ".join(
+            [
+                "tests/integration/ha_tests/scripts/deploy_chaos_mesh.sh",
+                namespace,
+            ]
+        ),
+        shell=True,
+        env=env,
+    )
+
+
+def destroy_chaos_mesh(namespace: str) -> None:
+    """Deploy chaos mesh to the provided namespace.
+
+    Args:
+        namespace: The namespace to deploy chaos mesh to
+    """
+    env = os.environ
+    env["KUBECONFIG"] = os.path.expanduser("~/.kube/config")
+
+    subprocess.check_output(
+        f"tests/integration/ha_tests/scripts/destroy_chaos_mesh.sh {namespace}",
+        shell=True,
+        env=env,
+    )
```

Comment:
Taken from the [mysql-k8s charm](https://github.com/canonical/mysql-k8s-operator/pull/116).